sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/retryprocessingerror/RetryProcessingErrorMLS.java

```diff
@@ -21,18 +21,57 @@
 
 /**
  * {@link MessageListenerStrategy} commits records for each partition. In case of processing errors
- * an error handler can decide if processing should be retried or stopped.
+ * the message will be retried for a configured amount of times.
+ *
+ * <p>After each error an error handler can decide if processing should be retried or stopped.
+ *
+ * <p>If the retry count exceeds:
+ *
+ * <ul>
+ *   <li>The failed message handling will be logged as ERROR
+ *   <li>a retryLimitExceededErrorHandler will be called allowing to do more error handling (i.e.
+ *       write to DLT)
+ *   <li>the offset will get increased, so the message handling gets ignored.
```

Comment:
Isn't there an exception (no offset increase) if the `retryLimitExceededErrorHandler` decides to stop processing?

Reply:
Technically you are right, the offset is conditional and will not be increased, if the `retryLimitExceededErrorHandler` throws an exception or returns `false`. For me it's part of the "a retryLimitExceededErrorHandler will be called allowing to do more error handling"